### PR TITLE
Contact Merge - Remove legacy call to old Rollups from Household Trigger

### LINKED ISSUE #4184: Contact Merge Still Queries for User-Defined Rollups Fields After Customizable Rollups are Enabled
If a customer deletes a field previously referenced by User Defined Rollups and they have Customizable Rollups enabled, they receive the following error during Contact Merge:

**First error: Invalid field: 'Opportunity.Amount_This_Year__c'**
(where _Amount This Year_ is a custom field I created and deleted).

*Stacktrace:*
```
Failed to invoke future method 'public static void rollupContactsandHouseholdsForTriggerFuture(Set<Id>)' on class 'npsp.RLLP_OppRollup' for job id 'xxxxxxxxxxx'
caused by: 
System.QueryException: Invalid field: 'Opportunity.Amount_This_Year__c'

Class.npsp.RLLP_OppRollup.rollupContacts: line 525, column 1
Class.npsp.RLLP_OppRollup.rollupContactsandHouseholdsForTrigger: line 163, column 1
Class.npsp.RLLP_OppRollup.rollupContactsandHouseholdsForTriggerFuture: line 132, column 1
```

### Steps to Repeat

1. Create field on Opportunity
    1. Name = Amount This Year
    2. Type = Formula (Currency)
    3. Formula: IF(YEAR( TODAY()), Amount, 0)
2. Create field on Contact
    1. Name = Total Amount This Year
    2. Type = Currency (16,2)
3. Create User Defined Rollup in NPSP Settings → Donations → User Defined Rollups
    1. Opportunity Field = Amount This Year
    2. Rollup Operation = SUM
    3. Target Object = Contact
    4. Target Field = Total Amount This Year
    5. Create Rollup
4. Enable Customizable Rollups
5. Create a Contact and a Closed Won Opportunity for that Contact.
6. Edit the Rollup using the NPSP Settings User Interface and switch to the following values:
    1. Change Operation to Sum
    2. Change Time Frame to Years Ago and Years Ago to This Year.
    3. Make Sure the Date Field = Close Date
    4. Change the Amount field to Amount
    5. Hit Save.
7. Manually delete the Rollup record in Setup → Custom Metadata Types → Rollups
8. Delete the Opportunity field called Amount This Year.
9. Create a second Contact with the same name.
10. Merge the 2 Contacts using the NPSP Contact Merge object.
11. Confirm the following error in NPSP Settings → System Tools → Error LogFirst error: Invalid field: 'Opportunity.Amount_This_Year__c'

### Workaround:
Go to Setup → Custom Settings → User Rollup Field Settings and delete the record that references the field you deleted.

### PR DESCRIPTION
**Pull request by force2b**
# Critical Changes

# Changes

- When Customizable Rollups is enabled in an org prevent the Household Trigger from executing legacy rollups on the merged Contact.

# Issues Closed

- Fixes #4184 

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by force2b** [comment]
**lurch: attach W-031664

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000008LlJSQA0" target="_blank">W-031664</a>

File: src/classes/HH_Households_TEST.cls [hunk 1]
```diff
@@ -575,44 +575,56 @@ public with sharing class HH_Households_TEST {
...
     /*********************************************************************************************************
-    * @description Tests merging a contact into a Household updates rollups on the Household
+    * @description Tests merging a contact into a Household updates rollups on the Household when usin gthe
+    * One-to-One Account Model
     */
-    static testMethod void testMergeRollupSupport(){
-        npe01__Contacts_and_Orgs_Settings__c contactSettingsForTests = UTIL_CustomSettingsFacade.getContactsSettingsForTests(new npe01__Contacts_and_Orgs_Settings__c (npe01__Account_Processor__c = CAO_Constants.ONE_TO_ONE_PROCESSOR, npe01__Enable_Opportunity_Contact_Role_Trigger__c = true));
-        npo02__Households_Settings__c householdSettingsForTests = UTIL_CustomSettingsFacade.getHouseholdsSettingsForTests(new npo02__Households_Settings__c (npo02__Household_Rules__c = HH_Households.ALL_PROCESSOR, npo02__Advanced_Household_Naming__c = true, npo02__Enable_Opp_Rollup_Triggers__c = false));
-         
+    /*static testMethod void testMergeRollupSupport(){
```
Comment (RadSFDeveloper) [inline src/classes/HH_Households_TEST.cls]: Let's delete the commented out unit test from the class.

Comment (force2b) [reply src/classes/HH_Households_TEST.cls]: Thanks. I forgot this has been commented out in my initial proof of concept. Might go back to enable this with some other related changes.

**Comment by RadSFDeveloper** [comment]
@Julian88Tex , the PR is ready for testing.

**Comment by Julian88Tex** [comment]
@rponti-sforg  @skristem  Automated tests have failures for this PR. Is this something I would typically alert the engineer to or automation or both? And/or do we usually ignore some of these failures?

**Comment by rponti-sforg** [comment]
> @rponti-sforg @skristem Automated tests have failures for this PR. Is this something I would typically alert the engineer to or automation or both? And/or do we usually ignore some of these failures?

Alert dev that is assigned to WI / PR.  Typically, they will kick off another build run and see if it passes then.  They usually do, but sometime they do not and further investigation is required.

**Comment by force2b** [comment]
Thanks @rponti-sforg.
@Julian88Tex As much as we love the builds, the do sometimes fail. The non-robot builds may fail with a lock error. Those are intermittent and we just click Rebuild to try again. If the non-robot builds fail with an assertion error, that's something we'd want to review. Pretty much anything else just gets rebuilt.

Robot tests may also fail for no particularly good reason (not there isn't the possibility that we broke something). We also rebuild those to see what happens.

**Comment by Julian88Tex** [comment]
@rponti-sforg @force2b Thanks this is all good to know! I'm guessing I can click rebuild as well... in github or meta?

**Comment by force2b** [comment]
@Julian88Tex The builds are re-running because of a merge into master from earlier. You can ignore the prerelease failure as long as it refers to `c:utilStencil`. We'll have that one for another week or so until the root issue is fixed. Ideally the other 3 will all pass this time.

**Comment by Julian88Tex** [comment]
@force2b hm, looks like Feature Test failed with a number of internal errors and perhaps one legit one:

`
Class.CRLP_FiscalYears.__sfdc_fiscalYearInfo: line 97, column 1
Class.CRLP_FiscalYears_TEST.testGetFiscalYearForDate: line 109, column 1

Message:
System.QueryException: List has no rows for assignment to SObject`

seems like we should run again although not aure how to trigger it

**Comment by force2b** [comment]
Totally strange @Julian88Tex at least for the internal errors. These did not exist before so I doubt it's an issue in the code. Rebuilding everything again.

**Comment by Julian88Tex** [comment]
@force2b branch has been tested, merged, and deleted

WI: https://foundation.lightning.force.com/lightning/r/agf__ADM_Work__c/a2x8000000AJfaeAAD/view

**Pull request merged**